src/sentry/api/endpoints/project_metrics_extraction_rules.py

```diff
@@ -85,11 +86,29 @@ def delete(self, request: Request, project: Project) -> Response:
 
         with handle_exceptions():
             with transaction.atomic(router.db_for_write(SpanAttributeExtractionRuleConfig)):
-                delete_extraction_rule_config(project, config_update)
-                schedule_invalidate_project_config(
-                    project_id=project.id, trigger="span_attribute_extraction_configs"
-                )
                 for config in config_update:
+                    try:
+                        span_attribute = SpanAttributeExtractionRuleConfig.objects.get(
+                            project=project, span_attribute=config["spanAttribute"]
+                        )
+                    except SpanAttributeExtractionRuleConfig.DoesNotExist:
+                        # ignore non existing extraction rules
+                        continue
```

Comment:
Does this actually happen or are we trying to please mypy here 😅  If it does happen this request should fail with either 400 or 404.

Reply:
Or at least, lets do a `logger.error` so that we get pinged if this happens

Reply:
I think we are trying to be idempotent here, and allow "deletion" as many times as it is requested.
Simon has implemented this, and there is a test that checks this, so I assumed that there is a reason behind it.

@shellmayr should we handle these differently or leave it as it is?

Reply:
For posterity: we talked about this and the right way to go is to make this idempotent, return 204 in line with the HTTP RFC, but log any instance of this in order to know when e.g. some front-end change starts sending too many requests with the same payload.